src/core/include/units/reference.h

```diff
@@ -0,0 +1,108 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2018 Mateusz Pusz
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#pragma once
+
+#include <units/bits/basic_concepts.h>
+#include <units/bits/dimension_op.h>
+
+namespace units {
+
+/**
+ * @brief The type for unit constants
+ *
+ * This type is intended to be used in the unit constants definition:
+ *
+ * @code{.cpp}
+ * namespace unit_constants {
+ * 
+ * inline constexpr auto m = reference<dim_length, metre>{};
+ * inline constexpr auto km = reference<dim_length, kilometre>{};
+ * 
+ * }
+ * @endcode
+ * 
+ * Unit constants simplify quantity creation:
+ * 
+ * @code{.cpp}
+ * using namespace units::isq::si::unit_constants;
+ *
+ * auto d = 123 * m;
+ * auto v = 70 * km / h;
```

Comment:
The more I think about this the more I prefer the syntax of:

```cpp
auto v = 70 * (km / h);
```

This nicely corresponds to:

```cpp
inline constexpr auto km_per_h = km / h;
auto v = 70 * km_per_h;
```

Reply:
That certainly looks nice here. But I wonder about larger formulas. Say you already have the `dist` as a `Length` variable. How do you do `dist / h` for `Speed`? I imagine you'd run into this when porting the glide computer example. I'd prefer to leave this out for another PR.

Reply:
This is exactly what I want to limit here. I think that an `lvalue * reference` should not compile (`dist / h`). It looks bad and is misleading. People will expect `h` to be a variable and not a constant and will look everywhere for it. I would prefer to type `dist / (1 * h)` instead to be explicit.

Reply:
So a multiplication operator on a `reference` should only take a value or another `reference` as `lhs`. `quantity` should not be allowed. Also `h * 1` should not be allowed.

Reply:
Basically, let's start small with something that is easy to understand even if more verbose or if it breaks "multiply maths". We can extend the interface later on if needed (or asked by the ISO Committee).

Reply:
That's understandable. I'll do it after addressing all other comments and before updating the documentation.